src/xrpld/rpc/handlers/AccountObjects.cpp

```diff
@@ -283,6 +283,10 @@ doAccountObjects(RPC::JsonContext& context)
             return RPC::invalid_field_error(jss::marker);
     }
 
+    // check if dirIndex is valid
+    if (!dirIndex.isZero() && !ledger->read({ltDIR_NODE, dirIndex}))
+        return RPC::invalid_field_error(jss::marker);
```

Comment:
since `RPC::getAccountObjects` returning `false` would have the same result, I would move this check into the body of `RPC::getAccountObjects`; this will keep the checks (other than the `marker` format, in the code segment above) in that one function only, which is more robust.

Reply:
updated